pkg/commands/entrypoints/entrypoints.go

```diff
@@ -0,0 +1,251 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package entrypoints provides the functionality to determine the entrypoints for Guardian.
+package entrypoints
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/abcxyz/guardian/pkg/flags"
+	"github.com/abcxyz/guardian/pkg/git"
+	"github.com/abcxyz/guardian/pkg/terraform"
+	"github.com/abcxyz/guardian/pkg/util"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+	"golang.org/x/exp/maps"
+)
+
+var _ cli.Command = (*EntrypointsCommand)(nil)
+
+// allowedFormats are the allowed format flags for this command.
+var allowedFormats = map[string]struct{}{
+	"json": {},
+	"text": {},
+}
+
+type EntrypointsCommand struct {
+	cli.BaseCommand
+
+	directory string
+
+	flags.GitHubFlags
+	flags.RetryFlags
+
+	flagPullRequestNumber       int
+	flagDestRef                 string
+	flagSourceRef               string
+	flagSkipDetectChanges       bool
+	flagFormat                  string
+	flagFailUnresolvableModules bool
+
+	gitClient git.Git
+}
+
+func (c *EntrypointsCommand) Desc() string {
+	return `Determine the entrypoint directories to run Guardian commands`
+}
+
+func (c *EntrypointsCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <directory>
+
+	Determine the entrypoint directories to run Guardian commands.
+`
+}
+
+func (c *EntrypointsCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+
+	c.GitHubFlags.Register(set)
+	c.RetryFlags.Register(set)
+
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this plan run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "dest-ref",
+		Target:  &c.flagDestRef,
+		Example: "ref-name",
+		Usage:   "The destination GitHub ref name for finding file changes.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source-ref",
+		Target:  &c.flagSourceRef,
+		Example: "ref-name",
+		Usage:   "The source GitHub ref name for finding file changes.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:   "skip-detect-changes",
+		Target: &c.flagSkipDetectChanges,
+		Usage:  "Skip detecting file changes and run for all entrypoint directories.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "format",
+		Target: &c.flagFormat,
+		Usage:  fmt.Sprintf("The format to print the output directories. The supported formats are: %s.", strings.Join(maps.Keys(allowedFormats), ", ")),
```

Comment:
But actually, I suggest making `var allowFormatNames = maps.Keys(allowedFormats)` a global variable. Because you also probably want to add a predictor here with those values as a list.